### PR TITLE
Disable intel packages for macOS

### LINKED ISSUE #7396: Installation of intel-mkl under macOS
Installing intel-mkl under macOS gives the issue described here: #3730. I had some time to investigate it in detail. This is not an Intel issue. Intel provides different packages for Linux and macOS. Both contain the same ``install.sh``, but the rest of the packages are different. The macOS package is a ``*.dmg`` archive.

I can not see any way to get it to work under macOS. A workaround is as follows:
1. Download intel-mkl from Intel's homepage
2. Install intel-mkl manually
3. Add the following to your ``~/.spack/packages.yaml``:
```
all:
  intel-mkl:
    paths:
      intel-mkl@2018.1.126 arch=<your-system>: /opt/intel
```

### PR DESCRIPTION
**Pull request by f-sim**
The intel build process is not supported for macOS. A workaround is documented in the documentation.

This PR resolves #7396.

### CODE REVIEW

File: lib/spack/docs/known_issues.rst [hunk 1]
```diff
@@ -88,3 +88,19 @@ software outside of Spack. Unfortunately, this command no longer works.
 See https://github.com/spack/spack/issues/2597 and
 https://github.com/spack/spack/issues/2662 for details. This is expected
 to be fixed by https://github.com/spack/spack/pull/2664.
+
+--------------------------------
+intel-mkl does not work on macOS
```
Comment (davydden) [inline lib/spack/docs/known_issues.rst]: work-> install (it does work)

File: lib/spack/docs/known_issues.rst [hunk 2]
```diff
@@ -88,3 +88,19 @@ software outside of Spack. Unfortunately, this command no longer works.
...
+--------------------------------
+
+**Status:** Only a workaround available due to the way Intel provides
+the library
+
+Installing intel-mkl under macOS requires a different download. This
+is not compatible with spack. However, there exists a workaround which
```
Comment (davydden) [inline lib/spack/docs/known_issues.rst]: Spack

File: var/spack/repos/builtin/packages/intel-mkl/package.py [hunk 3]
```diff
@@ -67,10 +67,6 @@ class IntelMkl(IntelPackage):
     provides('scalapack')
     provides('mkl')
 
-    if sys.platform == 'darwin':
```
Comment (davydden) [inline var/spack/repos/builtin/packages/intel-mkl/package.py]: Please keep that one, this is a different issue

**Comment by f-sim** [comment]
Can you give me some hint how to add tests for the new code?

File: lib/spack/docs/known_issues.rst [hunk 4]
```diff
@@ -88,3 +88,19 @@ software outside of Spack. Unfortunately, this command no longer works.
...
+is as follows:
+
+1. Download the intel-mkl version for macOS from Intel's homepage.
+2. Install the downloaded version manually.
```
Comment (mgsternberg) [inline lib/spack/docs/known_issues.rst]: There are a few additional wrinkles with a Mac installation.  You may want add the following caveat to the doc here:

-----
MKL _can not_ be installed as regular user – administrator privileges are required.

Moreover, the user running the MKL GUI installer is assumed to either (a) be an administrator (able to `sudo`), or (b) has set a password for `root` on the local machine (which is not a given on Mac).

If neither is the case, choose one of the following:
* Log in as administrator, then run the m_mkl*.app normally,
**or**
* Run the following shell command directly:
```
sudo /Volumes/m_mkl_2018.1.126/m_mkl_2018.1.126.app/Contents/MacOS/install.sh
```
(Adapt the version numbers as appropriate.)

-----

Here's the evidence: Starting the GUI installer as regular user brings up the following:
screen shot 2018-03-17 at 14 56 02
Observe how there is no means to supply an admin _user name_ as would normally be done if the system-native elevation dialog were used – sigh. Of lesser importance is the following, though also a bit of an eye-roller:
```
$ ./install.sh --cli-mode

Graphical user interface must be enabled to run this program.

Quitting!
```

Comment (davydden) [reply lib/spack/docs/known_issues.rst]: @f-sim could you please add this comment?

**Comment by alalazo** [comment]
Closing as stale. Feel free to reopen if you want to push this forward.

**Pull request closed without merging**